tests/intl/intltest.cpp

```diff
@@ -526,6 +526,124 @@ TEST_CASE("wxLocaleIdent::FromTag", "[uilocale][localeident]")
+    CHECK(FindBestMatch({ "zh" }, { "zh-CN", "zh-TW" }, "en-GB") == "zh-CN");
+    CHECK(FindBestMatch({ "zh-Hans-CN" }, { "zh-CN", "zh-TW" }, "en-GB") == "zh-CN");
+    CHECK(FindBestMatch({ "zh-Hant-HK" }, { "zh-CN", "zh-TW" }, "en-GB") == "zh-TW");
+
+    // language-specific script fallbacks
+    CHECK(FindBestMatch({ "sr" }, { "en", "sr-Latn" }, "en-GB") == "sr-Latn");
+    CHECK(FindBestMatch({ "sr-Cyrl" }, { "en", "sr-Latn" }, "en-GB") == "sr-Latn");
+
+    CHECK(FindBestMatch({ "de", "en-US"}, {"fr", "en-GB", "ja", "es-ES", "es-MX"}, "en-US") == "en-GB");
+    CHECK(FindBestMatch({ "de", "zh" }, { "fr", "en-GB", "ja", "es-ES", "es-MX" }, "en-US") == "en-US");
+
+    // match on maximized
+    CHECK(FindBestMatch({ "ja-JP", "en-GB" }, { "fr", "en-GB", "ja", "es-ES", "es-MX" }, "en-US") == "ja");
+    CHECK(FindBestMatch({ "ja-Jpan-JP", "en-GB" }, { "fr", "en-GB", "ja", "es-ES", "es-MX" }, "en-US") == "ja");
+    CHECK(FindBestMatch({ "zh", "en" }, { "fr", "zh-Hant", "en" }, "en-GB") == "en");
+
+    // close enough match on maximized
+    CHECK(FindBestMatch({ "de-CH", "fr" }, { "en-GB", "en", "de", "fr", "ja" }, "en-GB") == "de");
+    CHECK(FindBestMatch({ "en-US", "ar", "nl", "de", "ja" }, { "en-GB", "en", "de", "fr", "ja"}, "en-GB") == "en");
+
+    // best matches for Portuguese
+    CHECK(FindBestMatch({ "pt-PT", "es", "pt" }, { "pt-PT", "pt-BR", "es", "es-AR"}, "en-GB") == "pt-PT");
+    CHECK(FindBestMatch({ "pt-PT", "es", "pt" }, { "pt-PT", "pt", "es", "es-AR" }, "en-GB") == "pt-PT");
+    CHECK(FindBestMatch({ "pt-PT", "es", "pt" }, { "pt-BR", "es", "es-AR" }, "en-GB") == "pt-BR");
+
+    CHECK(FindBestMatch({ "pt", "es", "pt-PT" }, { "pt-PT", "pt-BR", "es", "es-AR" }, "en-GB") == "pt-BR");
+    CHECK(FindBestMatch({ "pt", "es", "pt-PT" }, { "pt-PT", "pt", "es", "es-AR" }, "en-GB") == "pt");
+    CHECK(FindBestMatch({ "pt", "es", "pt-PT" }, { "pt-BR", "es", "es-AR" }, "en-GB") == "pt-BR");
+    CHECK(FindBestMatch({ "pt-US", "pt-PT" }, { "pt-PT", "pt-BR", "es", "es-AR" }, "en-GB") == "pt-BR");
+    CHECK(FindBestMatch({ "pt-US", "pt-PT" }, { "pt-PT", "pt", "es", "es-AR" }, "en-GB") == "pt");
+
+    // regional specials
+    CHECK(FindBestMatch({ "en-AU" }, { "en", "en-GB", "es-ES", "es-AR" }, "de") == "en-GB");
+    CHECK(FindBestMatch({ "es-MX" }, { "en", "en-GB", "es-ES", "es-AR" }, "de") == "es-AR");
+    CHECK(FindBestMatch({ "es-PT" }, { "en", "en-GB", "es-ES", "es-AR" }, "de") == "es-ES");
+
+    // best match for traditional chinese
+    CHECK(FindBestMatch({ "zh-TW" }, { "fr", "zh-Hans-CN", "en-US" }, "en-GB") == "en-GB");
+    CHECK(FindBestMatch({ "zh-Hant" }, { "fr", "zh-Hans-CN", "en-US" }, "en-GB") == "en-GB");
+
+    CHECK(FindBestMatch({ "zh-TW", "en" }, {"fr", "zh-Hans-CN", "en-US"}, "en-GB") == "en-US");
+    CHECK(FindBestMatch({ "zh-Hant-CN", "en" }, {"fr", "zh-Hans-CN", "en-US"}, "en-GB") == "en-US");
+    CHECK(FindBestMatch({ "zh-Hans", "en" }, {"fr", "zh-Hans-CN", "en-US"}, "en-GB") == "zh-Hans-CN");
```

Comment:
All these tests should be (instead of or in addition too) having just plain `zh-Hans` or `zh-Hant` (and also *both*) in the supported list — that is by far the more common situation that having full locales there.

Reply:
ok, I'll adjust the test cases accordingly.